feathr_project/feathr/feature.py

```diff
@@ -1,13 +1,20 @@
-from abc import ABC, abstractmethod
+import re
 from copy import deepcopy
 from typing import List, Optional, Union, Dict
 
 from jinja2 import Template
 
 from feathr.dtype import FeatureType
-from feathr.transformation import ExpressionTransformation, Transformation, WindowAggTransformation
-from feathr.typed_key import DUMMY_KEY, TypedKey
 from feathr.frameconfig import HoconConvertible
+from feathr.transformation import ExpressionTransformation, Transformation
+from feathr.typed_key import DUMMY_KEY, TypedKey
+
+
+class FeatureNameValidationError(ValueError):
```

Comment:
I could not find a standard feathr exception base class when browsing the code, so I created this new exception. Happy to just raise a ValueError or somesuch as well.

Reply:
Done